src/azul/service/manifest_service.py

```diff
@@ -815,15 +824,39 @@ def manifest_config(self) -> ManifestConfig:
                               combine_script='return new ArrayList(params._agg.fields)',
                               reduce_script=reduce_script)
         es_search = es_search.extra(size=0)
-        response = es_search.execute()
-        assert len(response.hits) == 0
+        try:
+            response = es_search.execute()
+        # If timeout is exceeded the request is partitioned to reduce load on ES
+        except ConnectionTimeout:
+            fields = self._partition_search(es_search)
+        else:
+            assert len(response.hits) == 0
+            fields = response.aggregations.fields.value
         return {
             'contents': {
                 value: value.split('.')[-1]
-                for value in sorted(response.aggregations.fields.value)
+                for value in sorted(fields)
             }
         }
 
+    def _partition_search(self, es_search: Search) -> Set[str]:
+        requests = [es_search.query(Q('bool',
+                                      must=Q('prefix',
+                                             **{'bundles.uuid.keyword': p})))
+                    for p in map(''.join, product('0123456789abcdef'))]
+        logger.info('Sending asynchronous requests to Elasticsearch for manifest config')
+        start = time.time()
+        with ThreadPoolExecutor(max_workers=(len(requests) // 2)) as tpe:
```

Comment:
For other use of a TPE in the codebase i dont see `(len() // 2)` usually its just 1 thread per object in the sequence. Is this significant?

Reply:
I doesn't make a noticeable difference, I thought I use 8 workers instead of 16 to not overload Elasticsearch.

Reply:
I think you might want to talk to @hannes-ucsc about this, your original implementation could be better at not hammering down ES.